dlite_cuds/strategies/parse.py

```diff
@@ -81,16 +81,29 @@ def get(
         Returns:
             key to CUDS object in cache.
         """
-        cache = DataCache(self.parse_config.configuration.datacache_config)
-
+        # cache = DataCache(self.parse_config.configuration.datacache_config)
+        cache = DataCache()
         # Get CUDS-file and dump list of triples in cache
-        downloader = create_strategy("download", self.parse_config)
-        cudsfile = downloader.get()
-        cuds_key = cudsfile["key"]
-
+        # downloader = create_strategy("download", self.parse_config)
+        # cudsfile = downloader.get()
+        # cuds_key = cudsfile["key"]
+        config = self.parse_config.configuration
+        cacheconfig = config.datacache_config
+        if cacheconfig and cacheconfig.accessKey:
+            cuds_key = cacheconfig.accessKey
+        elif session and "key" in session:
+            cuds_key = session[
+                "key"
+            ]  # Is key overwritten every time by the download strategy?
+
+        else:
+            raise ValueError(
+                "either `location` or `datacache_config.accessKey` must be provided"
+            )
+        # Is key overwritten every time by the download strategy?
         # Get Ontology-file and dump list of triples in cache
         onto_download_config = ResourceConfig(
-            downloadUrl=self.parse_config.configuration.ontologyUrl,
+            downloadUrl=config.ontologyUrl,
             mediaType="application/CUDS",
         )
         downloader = create_strategy("download", onto_download_config)
```

Comment:
This is probably fine. An alternative implementation (that probably is more in line with the original design of oteapi) would be to have a separate download strategy before that parse strategy that downloads the ontology. Ex:

    download_ontology >> download_cuds >> parse_cuds

If doing this, we have to make sure that the two download strategies uses different keys when storing into the data cache.

Reply:
Sure, but this is not implemented well yet. But I agree on the principle. I suggest we continue with this for now.